source/client/lsp-raw_clients.adb

```diff
@@ -325,12 +325,24 @@ package body LSP.Raw_Clients is
                   Start := Index (Client.Buffer, New_Line & New_Line);
 
                   if Start /= 0 then
-                     Parse_Headers
-                       (Slice (Client.Buffer, 1, Start + 1),
-                        Client.To_Read);
-
-                     Delete
-                       (Client.Buffer, 1, Start + 2 * New_Line'Length - 1);
+                     begin
+                        Parse_Headers
+                          (Slice (Client.Buffer, 1, Start + 1),
+                           Client.To_Read);
+                        Delete
+                          (Client.Buffer, 1, Start + 2 * New_Line'Length - 1);
+                     exception
+                        when E : others =>
```

Comment:
Can't we be more precise regarding the exception we expect?

Reply:
Parse_Headers is raising Constraint_Error for now, I can add a custom Exception if needed